internal/bminventory/inventory.go

```diff
@@ -3798,14 +3798,20 @@ func (b *bareMetalInventory) DownloadMinimalInitrd(ctx context.Context, params i
 	var netFiles []staticnetworkconfig.StaticNetworkConfigData
 	var scriptContent, serviceContent string
 	if infraEnv.StaticNetworkConfig != "" {
+		var includesMACIdentifier bool
+		includesMACIdentifier, err = b.staticNetworkConfig.CheckConfigForMACIdentifier(infraEnv.StaticNetworkConfig)
+		if err != nil {
+			return common.GenerateErrorResponder(err)
+		}
+
 		// backward compatibility - nmstate.service has been available on RHCOS since version 4.14+, therefore, we should maintain both flows
 		var ok bool
 		ok, err = b.staticNetworkConfig.NMStatectlServiceSupported(infraEnv.OpenshiftVersion)
 		if err != nil {
 			return common.GenerateErrorResponder(err)
 		}
 
-		if ok {
+		if ok && !includesMACIdentifier {
```

Comment:
As we have the same logic in lines 6012-6024, consider extracting it to an helper func.
I.e. add both of these checks to a func named something like: 'shouldUseNmstatectlService'